client/state/preferences/actions.js

```diff
@@ -53,17 +70,29 @@ export const setPreference = ( key, value ) => ( {
  */
 export const savePreference = ( key, value ) => dispatch => {
 	dispatch( setPreference( key, value ) );
+	dispatch( {
+		type: PREFERENCES_SAVE,
+		key,
+		value
+	} );
+
 	const settings = {
 		[ USER_SETTING_KEY ]: {
 			[ key ]: value
 		}
 	};
 
-	return wpcom.me().settings().update( JSON.stringify( settings ) )
-		.then( ( data ) => {
-			dispatch( {
-				type: PREFERENCES_RECEIVE,
-				data
-			} );
+	return wpcom.me().settings().update( settings ).then( ( data ) => {
```

Comment:
>Do we need the `JSON.stringify` here?

Ah yes, this is ringing bells now. At a glance it doesn't appear you need it, but it doesn't preserve types when not used. IIRC, it's effectively the difference between saving as:

```
calypso_preferences[foo]=true
```

and

```
calypso_preferences={"foo":true}
```

Where the former treats `true` as a string server-side. When passed as a JSON string, the server-side will parse it and preserve types.